lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2788,3 +2788,38 @@ bool swift::diagnoseDeclAvailability(const ValueDecl *Decl,
   AvailabilityWalker AW(TC, DC);
   return AW.diagAvailability(const_cast<ValueDecl *>(Decl), R, nullptr, Flags);
 }
+
+void swift::validateExplicitIntroVersion(Decl *decl) {
+  // Check only if the command line option was set.
+  if (!decl->getASTContext().LangOpts.RequireExplicitIntroVersion)
+    return;
+
+  // Skip nominal type members as the type should be annotated.
+  auto declContext = decl->getDeclContext();
+  if (isa<NominalTypeDecl>(declContext))
```

Comment:
That's a good point, but I'm worried about requiring code this verbose. This would mean 4 lines of availability per declaration in some frameworks, and writing availability attributes on each case of an enum.

The issue of a new member being added to the ABI without an availability annotation should be detected by the ABI checker. I'm not sure what is more convenient, asking for explicit availability of all decls or relying on later detection of this mistake by the ABI checker.

Reply:
We've talked about this off GitHub and I think this way makes sense, at least for now.